piptools/repositories/base.py

```diff
@@ -12,24 +15,26 @@ def freshen_build_caches(self):
         """Should start with fresh build/source caches."""
 
     @abstractmethod
-    def find_best_match(self, ireq):
+    def find_best_match(
+        self, ireq: InstallRequirement, prereleases: Optional[bool]
```

Comment:
Could we default `preeleases` to `False` rather than making it optional?

Reply:
Usually, prereleases goes to [SpecifierSet.filter()](https://github.com/pypa/pip/blob/bbf8466088655d22cd46b286c8f0b8150754c1d9/src/pip/_vendor/packaging/specifiers.py#L816) where it's optional for a special case. Do you still think we should drop `Optional`?

Reply:
I see what you're saying. Good point. Lets match the upstream signature, so keep it optional. Thanks for explaining.